### PR TITLE
Enable json path syntax on keys

### PR DESCRIPTION
**Pull request by sbuljac**
Any thoughts on this?

### CODE REVIEW

File: src/conditions/EqualsCondition.ts [hunk 1]
```diff
@@ -26,8 +26,10 @@ export class EqualsCondition implements IConditionFunction {
 
         return Object.keys(args).every((key) => {
             return CommonUtil.matchesAnyElement(args[key], (elm) => {
-                return context[key] && ConditionUtil.getValueByPath(context, elm) === context[key];
+                const keyValue = key.startsWith('$.') ?  ConditionUtil.getValueByPath(context, key) : context[key];
```
Comment (koladilip) [inline src/conditions/EqualsCondition.ts]: ConditionUtil.getValueByPath(context, key) -> this is going to give only key right. 
So shouldn't this be:
const keyValue = context[ConditionUtil.getValueByPath(context, key)];

Comment (sbuljac) [reply src/conditions/EqualsCondition.ts]: ConditionUtil.getValueByPath(context, key) will give value on context when key is in JSON path format, I need this so I can compare nested values, e.g.
```
{ '$.requester.employee.id': '$.instance.employee' }
```
or
```
{ '$.instance.creatorId': '$.instance.assigneeId' }
```

**Comment by Crookshanks16** [comment]
For our use case we would need that feature as well.

File: src/conditions/EqualsCondition.ts (hunk 1 above)
Comment (koladilip) [reply src/conditions/EqualsCondition.ts]: can you please give a full example with grants and context and conditions?
Also please update: https://github.com/tensult/role-acl/blob/develop/test/acl.spec.ts for this use case.

**Comment by koladilip** [comment]
https://github.com/tensult/role-acl/commit/b2ce6b959893fe08e0a00d06c1b874c64886eaa4

I have added test cases

**Pull request merged**